proxy/http/HttpConfig.cc

```diff
@@ -1120,6 +1125,20 @@ HttpConfig::startup()
   // Local Manager
   HttpEstablishStaticConfigLongLong(c.synthetic_port, "proxy.config.admin.synthetic_port");
 
+  // The responses with the following status code WILL BE cached with negative caching enabled.
+  c.codeNegCache.add(HTTP_STATUS_NO_CONTENT);
+  c.codeNegCache.add(HTTP_STATUS_USE_PROXY);
+  c.codeNegCache.add(HTTP_STATUS_FORBIDDEN);
+  c.codeNegCache.add(HTTP_STATUS_NOT_FOUND);
```

Comment:
Honestly, since the number of status codes are limited, a standard C array with 1000 bool elements would be easier, and definitely faster.

```
memzero(codeNegCache, sizeof(codeNegCache);
codeNegCache[HTTP_STATUS_NO_CONTENT] = true;
codeNegCache[HTTP_STATUS_USE_PROXY] = true;
```
etc. Then the check if a particular status code is cacheable is O(1). Yeh, so we waste 1KB (give or take) of RAM, but I bet all this code makes up for that anyways :-).